networkx/algorithms/graph_hashing.py

```diff
@@ -312,14 +407,28 @@ def weisfeiler_lehman_step(G, labels, node_subgraph_hashes, edge_attr=None):
             node_subgraph_hashes[node].append(hashed_label)
         return new_labels
 
+    if iterations <= 0:
+        raise ValueError(
+            "The WL algorithm is only meaningful when run for a positve number of iterations."
```

Comment:
maybe shorten the error message?  (tests for it would change too)
```suggestion
            "The WL algorithm requires that `iterations` be positive"
```